### DISCUSSION TITLE
findPath unsuccessful

### DISCUSSION THREAD

**Discussion by fax1ty** [isaac-mason/recast-navigation-js#371] [General]
vite-node:
```
Error: findPath unsuccessful
    at ...\server\src\index.ts:39:1
    at ViteNodeRunner.runModule (.../node_modules/vite-node/dist/client.mjs:362:5)
    at ViteNodeRunner.directRequest (.../node_modules/vite-node/dist/client.mjs:346:5)
    at ViteNodeRunner.cachedRequest (.../vite-node/dist/client.mjs:189:14)
    at ViteNodeRunner.executeFile (.../node_modules/vite-node/dist/client.mjs:161:12)
    at CAC.run (.../node_modules/vite-node/dist/cli.mjs:91:5)
```

bun:
```
error: findPath unsuccessful
      at ...\server\src\index.ts:36:1
```

## Additional context 

`.findClosestPoint()` returns

```js
{
  success: false,
  status: -2147483640,
  polyRef: 0,
  point: { x: 0, y: 0, z: 0 },
  isPointOverPoly: false
}
```

---
[console.log(RecastThree)](https://github.com/isaac-mason/recast-navigation-js/blob/main/examples/node-esm-example/index.mjs#L8) returns `undefined`

---
But `.findRandomPoint()` is working as expected. Returns 
```
{
  success: true,
  status: 1073741824,
  randomPolyRef: 12182,
  randomPoint: { x: -8.3472261428833, y: 0.20000000298023224, z: 95.33012390136719 }
}
```

Am I doing anything wrong?

**Comment by isaac-mason** [comment]
Could you share a codesandbox/repo with a reproduction of this issue? Happy to take a look 🙂

**Comment by fax1ty** [comment]
https://github.com/fax1ty/recast-findpath-repro-bun
https://github.com/fax1ty/recast-findpath-repro-vite

To clarify: this also doesn't work with `generateSoloNavMesh` when I pass raw geometry data

P.S. I don't know why, but `console.log(RecastThree)` is returning something now
(Maybe something with monorepo support? 🤨 Before, I had all the dependencies in the root and the projects in subdirectories)

**Comment by isaac-mason** [comment]
Hey @fax1ty, can you provide the code including the call to `computePath` / `findPath` that leads to the error?


Without seeing that I'm guessing here - but assuming that you're using `computePath`, the start / end positions may be too far away from the navmesh to find the nearest navmesh polys? If this were the case, you could try increasing `navMeshQuery.defaultQueryHalfExtents`?

Loading the navmesh export from the repo, and calling `computePath` with two positions I arbirarily picked is working ok.

```ts
    const { point: start } = navMeshQuery.findClosestPoint({
      x: 88.56768727667445,
      y: 0.20000000298023224,
      z: -102.46081100138719,
    });

    const { point: end } = navMeshQuery.findClosestPoint({
      x: -74.65151041753724,
      y: 0.20000000298023224,
      z: 111.50702247544152,
    });

    const { path } = navMeshQuery.computePath(start, end);
```

<img width="1009" alt="image" src="https://github.com/isaac-mason/recast-navigation-js/assets/67411435/6721e539-80a0-4265-800c-c4654c0e370c">

Also happy to take a look at your issue with `generateSoloNavMesh` if you can provide the code and model.

**Comment by fax1ty** [comment]
It started working starting from `navMeshQuery.defaultQueryHalfExtents = new Vector3(10000, 10000, 10000);` but returns only 2 points. Is it intended?
![image](https://github.com/isaac-mason/recast-navigation-js/assets/26363214/4f94d54e-86a0-4de5-8ae8-b93dc8efcc89)



How do I use it:
```ts
  moveTo(b: Vector3) {
    const a = this.actor.rigidbody.translation(); // rapier translation

    const { point: to, success: s1 } = this.navigation.findClosestPoint(b);
    if (!s1) return;

    const { point: from, success: s2 } = this.navigation.findClosestPoint(a);
    if (!s2) return;

    const { path, error } = this.navigation.computePath(from, to);
    if (error) return;
    this.path = path;
  }
  ```

**Comment by fax1ty** [comment]
Agents don't work either

```ts
    const {
      point: to,
      success,
      status,
    } = this.navigation.findClosestPoint(point);
    if (!success) throw new Error(statusToReadableString(status));

    const result = this.agent.requestMoveTarget(to);
    console.log(result); // -> false
```
    
```ts
    this.agent = crowd.addAgent(this.actor.rigidbody.translation(), {});
```
    
```ts
    const crowd = new Crowd(navigation, {
  maxAgents: Number.MAX_SAFE_INTEGER,
  maxAgentRadius: Number.MAX_SAFE_INTEGER,
});
```

**UPD:** It's ok with 
```ts
const crowd = new Crowd(navigation, {
  maxAgents: 10000,
  maxAgentRadius: 100,
});
```

**Comment by isaac-mason** [comment]
Hey,

Note that increasing the query extents increases the overhead of the operation, so it's best to use a value just big enough for your use case. 

Similarly, specifying a large max number of agents will allocate extra memory for each agent. That's why setting max agents to such a large value was causing issues.

As for your computePath question - if there's nothing in between the start and end points, then you'll only get two points. I can't help you without more information though. Without sharing a reproduction it's impossible to know whether you're running into issues with usage or with the library. 

If you haven't already, take a look through the examples 🙂
 https://github.com/isaac-mason/recast-navigation-js/tree/main/packages/recast-navigation/.storybook/stories
 https://recast-navigation-js.isaacmason.com/
